cli.js

```diff
@@ -1,21 +1,22 @@
 #!/usr/bin/env node
 
 'use strict';
-const childProcess = require('child_process');
-const isCi = require('is-ci');
 
-const isWindows = process.platform === 'win32';
-const npmExec = isWindows ? 'npm.cmd' : process.env.npm_execpath ? process.env.npm_execpath : 'npm';
+process.on('unhandledRejection', err => {
+	throw err;
+});
+
+const spawn = require('cross-spawn');
+const isCi = require('is-ci');
 
 function run(args, isCi) {
 	const script = isCi ? args[0] : args[1];
 
 	if (script) {
-		return childProcess.spawn(
-			npmExec,
+		return spawn(
+			'npm',
```

Comment:
checking `process.env.npm_execpath` would support running through `yarn` (or alternative npm-like things).

Reply:
I can't seem to find any documentation on `process.env.npm_execpath`, do you have a link?

Reply:
Can't find any great documentation around it, other than some discussion on its use:

https://github.com/yarnpkg/yarn/issues/5063#issuecomment-354794424

And code where npm and yarn are setting its value:
- npm: https://github.com/npm/cli/commit/ad8e21fc9da6e9a0c46f2c69f999e280e2e3d188
- yarn: https://github.com/yarnpkg/yarn/blob/a3b1294c22043cd8868b6a1decb77b673e764efd/src/util/execute-lifecycle-script.js#L81

I also checked `pnpm` and it supports this var too.

Reply:
Cool, will update it to make use of it. Should it be `node` with `npmExec` as an argument? (`node, [npmExec, 'run', script]`?)

Reply:
Yeah, that sounds good 👍